engine/src/main/java/org/terasology/world/block/BlockRegion.java

```diff
@@ -19,317 +19,384 @@
 import org.joml.Vector3ic;
 import org.terasology.entitySystem.entity.EntityRef;
 
+import java.util.Optional;
+
 /**
- * is a bounded box describing blocks contained within. A {@link BlockRegion} is described and backed by an {@link
- * AABBi}
+ * A bounded, axis-aligned volume in space denoting a collection of blocks contained within.
  */
 public class BlockRegion {
 
     /**
-     * AABB region that backs a BlockRegion
+     * The x coordinate of the minimum corner.
+     */
+    private int minX = Integer.MAX_VALUE;
+    /**
+     * The y coordinate of the minimum corner.
+     */
+    private int minY = Integer.MAX_VALUE;
+    /**
+     * The z coordinate of the minimum corner.
+     */
+    private int minZ = Integer.MAX_VALUE;
+    /**
+     * The x coordinate of the maximum corner.
      */
-    public final AABBi aabb = new AABBi();
+    private int maxX = Integer.MIN_VALUE;
+    /**
+     * The y coordinate of the maximum corner.
+     */
+    private int maxY = Integer.MIN_VALUE;
+    /**
+     * The z coordinate of the maximum corner.
+     */
+    private int maxZ = Integer.MIN_VALUE;
 
-    public BlockRegion() {
-    }
+    // -- CONSTRUCTORS -----------------------------------------------------------------------------------------------//
 
-    public BlockRegion(BlockRegion source) {
-        aabb.set(source.aabb);
+    /**
+     * Creates an empty block region with invalid minimum/maximum corners.
+     * <p>
+     * {@link #isValid()} will return {@code false} for an empty block region created via this constructor.
+     */
+    @Deprecated
+    BlockRegion() {
```

Comment:
This is not public anymore, so there is hopefully no way to create invalid BlockRegions anymore. However, this comes in very handy when redcuing a list of positions in the `encompassing` helper methods, that's why I left it in...

Not sure what to do with it...

Reply:
I wouldn't mark it as deprecated though.